routes/students.routes.js

```diff
@@ -0,0 +1,83 @@
+const router = require("express").Router()
+
+const User = require("../models/User.model")
+
+const {isLogged, isPM} = require("../utils")
+
+const { isLoggedIn, checkRoles } = require("../middleware")
+
+
+//Students List
+router.get('/', (req, res) => {
+    
+    User
+    .find()
+    .then(regUser => {
+        res.render('students/students', { regUser, isLogged: isLogged(req.session.currentUser), isPM: isPM(req.session.currentUser?.role) })
+    })
+    .catch(err => console.log(err))
+})
+
+//Students Profile
+router.get('/:id', isLoggedIn,  (req, res) => {
+    
+    const {id} = req.params
+
+    User
+    .findById(id)
+    .then(student => {
+        res.render('students/profile', { student, isLogged: true })
+    })
+    .catch(err => console.log(err))
+})
+
+//Student Edit Profile
+router.get('/:id/edit', checkRoles('PM'), (req,res) => {
+
+    const {id} = req.params
+
+    User
+    .findById(id)
+    .then(profile => {
+        res.render('students/edit',profile)
+        
+    })
+    .catch(err => console.log(err))
+})
+
+router.post('/:id/edit', checkRoles('PM'), (req,res) => {
+     
+      const {id} = req.params
+
+      const {username, profileImg, description } = req.body
+
+      
+
+    User
+    .findByIdAndUpdate(id, {username, profileImg, description }, {new: true})
+    .then(User => res.redirect('/'))
+    .catch(err => console.log(err))
+})
```

Comment:
Es fundamental que compruebes lo que vas haciendo paso a paso. Aquí te faltan console.logs a patadas, por eso no entiendes que problema tienes con el 404.

```suggestion
router.post('/:id/edit', checkRoles('PM'), (req,res) => {
     
      const {id} = req.params
      console.log(id) // ===> muestra ":id"
      const {username, profileImg, description } = req.body

      
    //por lo tanto estás buscando el usuario con id => :id , te da un error porque ese id no es valido
    User
    .findByIdAndUpdate(id, {username, profileImg, description }, {new: true})
    .then(User => res.redirect('/'))
    .catch(err => console.log(err))
})
```